chain/chain/src/chain.rs

```diff
@@ -3553,6 +3553,56 @@ impl<'a> ChainUpdate<'a> {
         })
     }
 
+    /// For all the outgoing receipts generated in block `hash` at the shards we are tracking
+    /// in this epoch,
+    /// save a mapping from receipt ids to the destination shard ids that the receipt will be sent
+    /// to in the next block.
+    /// Note that this function should be called after `save_block` is called on this block because
+    /// it requires that the block info is available in EpochManager, otherwise it will return an
+    /// error.
+    pub fn save_receipt_id_to_shard_id(
```

Comment:
I'd rename this function (as the current name suggests that we're storing a single receipt id to a single shard).

maybe: save_outgoing_receipts_to_shards (or to_destination_shards)